x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -155,4 +287,270 @@ export const RelevanceTuningLogic = kea<
       (schema: Schema): boolean => Object.keys(schema).length >= 2,
     ],
   }),
+  listeners: ({ actions, values }) => ({
+    initializeRelevanceTuning: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings/details`;
+
+      try {
+        const response = await http.get(url);
+        actions.onInitializeRelevanceTuning(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    getSearchResults: async (_, breakpoint) => {
+      await breakpoint(250);
+
+      const { engineName } = EngineLogic.values;
+      const { http } = HttpLogic.values;
+      const query = values.query;
+      const { search_fields: searchFields, boosts } = removeBoostStateProps(values.searchSettings);
+      if (!query) {
+        return actions.clearSearchResults();
+      }
```

Comment:
If we have an early return, I'd like us to do that as soon as possible before destructuring / calling other helpers
```suggestion
      const query = values.query;
      if (!query) return actions.clearSearchResults();

      const { engineName } = EngineLogic.values;
      const { http } = HttpLogic.values;
      const { search_fields: searchFields, boosts } = removeBoostStateProps(values.searchSettings);
```

Also changed if return to a single line per Scotty's previous requests